# MULTIPLE LINEAR REGRESSION

- Best fitting straight line through a set of data points.
- x = independent variable
- y = dependent variable; y = f(x)

Predicting future values
- Once have line can predict y for any x.
- What this doesn't specify is error.


Had to fix for UoEO
1. Wrong CSV file provided. Actual is here: https://www.kaggle.com/datasets/chowdeswarreddy/carscsv
2. User warning. Added feature_df as UoEO trained model with a DataFrame (with column names as featuer names), then tried to predict using NumPy array (without feature names),

In [9]:
import pandas
from sklearn import linear_model

df = pandas.read_csv("cars.csv")

X = df[['Weight', 'Volume']]
y = df['CO2']

regr = linear_model.LinearRegression()
regr.fit(X, y)

# Predict the CO2 emission of a car where the weight is 2300kg, and the volume is 1300cm3:

feature_df = pandas.DataFrame([[2300, 1300]], columns=['Weight', 'Volume'])
predictedCO2 = regr.predict(feature_df)
print(predictedCO2)

[107.2087328]


# Coefficient

The coefficient is a factor that describes the relationship with an unknown variable.
In this case, we can ask for the coefficient value of weight against CO2, and for volume against CO2. The answer(s) we get tells us what would happen if we increase, or decrease, one of the independent values.

In [10]:
print(regr.coef_)

[0.00755095 0.00780526]


The result array represents the coefficient values of weight and volume.

Weight: 0.00755095
Volume: 0.00780526

These values tell us that if the weight increases by 1kg, the CO2 emission increases by 0.00755095g.

And if the engine size (Volume) increases by 1 cm3, the CO2 emission increases by 0.00780526 g.

I think that is a fair guess, but let us test it!

We have already predicted that if a car with a 1300cm3 engine weighs 2300kg, the CO2 emission will be approximately 107g.

What if we increase the weight with 1000kg (from 2300 to 3300) what will be the CO2 emission?

Ans: 107.2087328 + (1000 * 0.00755095) = 114.75968

In [11]:
# Predict CO2 emmisison of a car with weight 3300Kg, and volume 1300cm3.

feature_df = pandas.DataFrame([[3300, 1300]], columns=['Weight', 'Volume'])
predictedCO2 = regr.predict(feature_df)
print(predictedCO2)

[114.75968007]


In [13]:
# Predict CO2 emmisison of a car with weight 1000Kg, and volume 1600cm3. (MX5)

feature_df = pandas.DataFrame([[1000, 1600]], columns=['Weight', 'Volume'])
predictedCO2 = regr.predict(feature_df)
print(predictedCO2)

[99.73407861]
